src/Univ2LpOracle.sol

```diff
@@ -221,8 +225,9 @@ contract UNIV2LPOracle {
         require(ts == block.timestamp);
 
         // Adjust reserves w/ respect to decimals
-        if (dec0 != uint8(18)) res0 = uint112(res0 * 10 ** sub(18, dec0));
-        if (dec1 != uint8(18)) res1 = uint112(res1 * 10 ** sub(18, dec1));
+        // TODO: is the risk of overflow here worth mitigating? (consider an attacker who can mint a token at will)
+        if (normalizer0 > 1) res0 = uint112(res0 * normalizer0);
```

Comment:
This saves some gas, but I think we should keep `dec0` and `dec1` (in addition to this) since they provide some introspection on how `k` is calculated.

Reply:
Why can't people just read them from the source tokens? I don't see any value-add to having them here.

Reply:
Agree with @kmbarry1 I don't think we need them.